compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -178,30 +178,39 @@ void OMR::ValuePropagation::initialize()
    int32_t numParms = comp()->getMethodSymbol()->getParameterList().getSize();
    _parmInfo = (int32_t *) trMemory()->allocateStackMemory(numParms * sizeof(int32_t));
    _parmTypeValid = (bool *) trMemory()->allocateStackMemory(numParms * sizeof(bool));
-   // mark all parms as invariant (0) to begin with
-   //
-   memset(_parmInfo, 0, numParms * sizeof(int32_t));
+
    // mark info to be true
    for (int32_t i = 0; i < numParms; i++)
       _parmTypeValid[i] = true;
 
-   for (TR::TreeTop *tt = comp()->getStartTree();
-         tt;
-         tt = tt->getNextTreeTop())
+   if (comp()->isDLT())
```

Comment:
I would suggest a comment here explaining why DLT compiles are being treated separately. 

Should you have a loop that stores 1 into each index in _parmInfo ? I think the memset will store 1 into every byte of the array. I guess I am thinking of a case where some code checks if the _parmInfo int32_t value is precisely 1 and then goes wrong because it is a different non-zero value to the one that it expected.

Reply:
[See force-push](https://github.com/eclipse/omr/compare/034e0b5e42b5edd96228381351217ac576c798fe..6374ead44793a28753dd1816702448c0584057c1)